### PR TITLE
feat: add framework, build system, and package manager to telemetry for every command

### PR DESCRIPTION
**Pull request by sarahetter**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Closes https://github.com/netlify/pod-dev-foundations/issues/504

Adds the framework, build system and package manager to telemetry for further investigation into cli usage. Only successful runs of a command will be added to telemetry (as telemetry is only sent on base-command's `onEnd` function, which does not run if the process is ended with `process.exit(1)`) so we aren't capturing that information, but this is out of scope for this PR. 

---

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**
Screenshot 2023-05-25 at 2 03 29 PM

### CODE REVIEW

**Review by Skn0tt — DISMISSED** [review]
Looks good! If there's an easy way for us to add a unit test for this, that'd be great. If it's not obvious, then let's open an issue to track this.

File: src/commands/base-command.mjs [hunk 1]
```diff
@@ -456,6 +459,21 @@ export default class BaseCommand extends Command {
...
+    const frameworks = await project.detectFrameworks()
+
+    const frameworkIDs = frameworks?.map((framework) => framework.id)
+
+    if (frameworkIDs?.length !== 0) {
+      this.setAnalyticsPayload({ frameworks: frameworkIDs })
+    }
+
+    const buildSystem = await project.detectBuildSystem()
+    const packageManager = await project.detectPackageManager()
```
Comment (lukasholzer) [inline src/commands/base-command.mjs:474]: That's not needed as `detectFrameworks()` depends on it and run those first, so it just retrieves the value from the `project`.

you can do directly `{ packageManager: project.packageManager?.name, buildSystem: project.buildSystems.map(({id}) => id) }`

Comment (lukasholzer) [reply src/commands/base-command.mjs:474]: CleanShot 2023-05-26 at 08 53 46

**Review by lukasholzer — APPROVED** [review]
LGTM!

**Pull request merged**